### DISCUSSION TITLE
Too many decimal places

### DISCUSSION THREAD

**Discussion by Feridoun** [BottlecapDave/HomeAssistant-OctopusEnergy#524] [Q&A]
Hi, I've got target rate set up for use on an esp32 display above my washing machine. On the display I am showing the 'Next average cost' attribute so I can decide whether it's worth running it now or later. The display shows the next best time and average cost as a string but it's got way too many decimal places, e.g. 
Next average cost: 16.148999999999997p
Shall I just shorten the string to the first 2 characters on the ESP32, or could the data itself be changed to a float variable in home assistant? thanks. I am a novice at all of this.

**Comment by BottlecapDave** [comment]
This variable will be changing to a numeric type in the next release. For the time being you could do a substring via a template, then cast to float, then round to the number of decimals you need.

```
{{ state_attr('binary_sensor.octopus_energy_target_washing_machine', 'next_average_cost')[:-1] | float | round(2) }}
```

https://www.home-assistant.io/docs/configuration/templating/#numeric-functions-and-filters

**Comment by Feridoun** [comment]
That's really helpful thanks. I hope to use numeric format to apply conditional formatting.
